### PR TITLE
[CSL 341]: Port getBrowseResultsByItemIds

### PR DESCRIPTION
**Pull request by gardensgreen**

### CODE REVIEW

File: src/modules/browse.js [hunk 1]
```diff
@@ -97,6 +97,105 @@ function createBrowseUrl(filterName, filterValue, parameters, userParameters, op
   return `${serviceUrl}/browse/${encodeURIComponent(filterName)}/${encodeURIComponent(filterValue)}?${queryString}`;
 }
 
+// Create query params from parameters and options
+function createQueryParams(parameters, options) {
```
Comment (esezen) [inline src/modules/browse.js:101]: A lot of code is repeating between this and  `createBrowseUrl` you can split the repeating part into a new function and call that from these two. There is an example in my PR.

File: src/modules/browse.js [hunk 2]
```diff
@@ -186,6 +285,73 @@ class Browse {
...
+    if (userParameters.userIp && typeof userParameters.userIp === 'string') {
+      headers['X-Forwarded-For'] = userParameters.userIp;
+    }
+
+    // Append user agent as 'User-Agent' if available
+    if (userParameters.userAgent && typeof userParameters.userAgent === 'string') {
+      headers['User-Agent'] = userParameters.userAgent;
+    }
+
+    return fetch(requestUrl)
```
Comment (esezen) [inline src/modules/browse.js:330]: You need headers here. You can take a look at `getBrowseResults`

File: src/modules/browse.js [hunk 3]
```diff
@@ -186,6 +285,73 @@ class Browse {
...
+        if (json.response && json.response.results) {
+          if (json.result_id) {
+            // Append `result_id` to each result item
+            json.response.results.forEach((result) => {
+              // eslint-disable-next-line no-param-reassign
+              result.result_id = json.result_id;
+            });
+          }
+
+          // this.eventDispatcher.queue('browse.getBrowseResultsByItemIds.completed', json);
```
Comment (esezen) [inline src/modules/browse.js:348]: You can remove this

**Comment by esezen** [comment]
Made a couple of comments, the rest looks awesome! Great job

**Review by esezen — APPROVED** [review]
LGTM! Great work @gardensgreen

**Review by sblaurock — APPROVED** [review]
Nice work here!

**Pull request merged**